opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -529,16 +529,13 @@ class LiveOilPvt
             const Evaluation& f = RsTable.eval(pSat, /*extrapolate=*/true) - Rs;
             const Evaluation& fPrime = RsTable.evalDerivative(pSat, /*extrapolate=*/true);
 
-            const Evaluation& delta = f/fPrime;
-
-            if (!Opm::isfinite(delta)) {
-                 std::stringstream errlog;
-                 errlog << "Obtain delta with non-finite value for Rs = " << Rs << " in the "
-                        << i << "th iteration during finding saturation pressure iteratively";
-                 OpmLog::problem("wetgas NaN delta value", errlog.str());
-                 OPM_THROW_NOLOG(NumericalProblem, errlog.str());
+            //If the derivative is zero Newton will not converge, so simply return our best guess instead
+            if (Toolbox::scalarValue(fPrime) == 0.0) {
+                return pSat;
```

Comment:
After some testing, it appears to me that the results returning the minimum pressure, using the following code, 
```
            // If the derivative is zero Newton will not converge,
            // so simply return the smallest value.
            if (Toolbox::scalarValue(fPrime) == 0.0) {
                //Find interval where derivative is zero
                const auto& table = saturationPressure_[regionIdx];
                size_t index = RvTable.findSegmentIndex(Toolbox::scalarValue(pSat));
                Scalar x0 = RvTable.xAt(index);
                Scalar x1 = RvTable.xAt(index+1);

                //Evaluate at end points and chose minimum
                const Evaluation y0 = table.eval(x0, /*extrapolate=*/true);
                const Evaluation y1 = table.eval(x1, /*extrapolate=*/true);
                if (std::abs(Toolbox::scalarValue(y0)) < std::abs(Toolbox::scalarValue(y1))) {
                    return y0;
                }
                else {
                    return y1;
                }
            }
```
gives identical results to just returning pSat. This, however, requires that the previously private function Tabulated1DFunction::findSegmentIndex(value) is made public. I'm open for suggestions on what is best here.